plugins/modules/aws_api_gateway_domain.py

```diff
@@ -231,43 +231,58 @@ def delete_domain(module, client):
 
 
 @AWSRetry.backoff(**retry_params)
-def get_domain_name(client, domain_name):
-    return client.get_domain_name(domainName=domain_name)
+def get_domain_name(client, domain_name, module):
+    try:
+        return client.get_domain_name(domainName=domain_name)
+    except (BotoCoreError, ClientError) as e:
+        module.fail_json_aws(e, msg="Failed to get api gateway domain.")
```

Comment:
It looks like this will fail the first time a ClientError is encountered rather than retrying. Generally the exception handling should happen outside the retry wrapper.

Reply:
That's a good point.   
In that case it is already covered by try/catch L127-134.  
So I'm going to revert it and review the other cases also.

Reply:
Ok, all client calls all already covered by try and catch :)  
So I am going to close this PR.

Reply:
Thanks for checking :smile: